router/batchrouter/asyncdestinationmanager/bing-ads/bingads.go

```diff
@@ -55,94 +55,152 @@ func generateClientID(user User, metadata Metadata) string {
-	scanner := bufio.NewScanner(textFile)
-	size := 0
-	for scanner.Scan() {
-		line := scanner.Text()
-		var data Data
-		err := json.Unmarshal([]byte(line), &data)
-		if err != nil {
-			return "", nil, nil, err
-		}
-
-		payloadSizeStat := stats.Default.NewTaggedStat("payload_size", stats.HistogramType, map[string]string{
-			"module":   "batch_router",
-			"destType": b.destName,
-		})
-		payloadSizeStat.Observe(float64(len(data.Message.List)))
-
-		size = size + len([]byte(line))
-		if int64(size) < common.GetBatchRouterConfigInt64("MaxUploadLimit", b.destName, 100*bytesize.MB) {
-			for _, uploadData := range data.Message.List {
-				clientId := generateClientID(uploadData, data.Metadata)
-				err = csvWriter.Write([]string{"Customer List Item", "", "", audienceId, clientId, "", "", "", "", "", "", "Email", uploadData.HashedEmail})
-			}
-			successJobIds = append(successJobIds, data.Metadata.JobID)
-		} else {
-			// ?? how to add test case for this
-			failedJobIds = append(failedJobIds, data.Metadata.JobID)
-		}
-
-	}
-	csvWriter.Flush()
+	err = csvWriter.Write([]string{"Customer List", "", audienceId, "", "", "", "", "", "", "", fileType, "", ""})
+	return csvFilePath, csvWriter, zipFilePath
+}
 
-	// Create the ZIP file and add the CSV file to it
+func convertCsvToZip(csvFilePath string, zipFilePath string, eventCount int) error {
+	csvFile, err := os.Open(csvFilePath)
 	zipFile, err := os.Create(zipFilePath)
+	if eventCount == 0 {
+		os.Remove(csvFilePath)
+		os.Remove(zipFilePath)
+		return nil
+	}
 	if err != nil {
-		return "", nil, nil, err
+		return err
 	}
 	defer zipFile.Close()
 
 	zipWriter := zip.NewWriter(zipFile)
 
 	csvFileInZip, err := zipWriter.Create(filepath.Base(csvFilePath))
 	if err != nil {
-		return "", nil, nil, err
+		return err
 	}
 
 	_, err = csvFile.Seek(0, 0)
 	_, err = io.Copy(csvFileInZip, csvFile)
 	if err != nil {
-		return "", nil, nil, err
+		return err
 	}
 
 	// Close the ZIP writer
 	err = zipWriter.Close()
 	if err != nil {
-		return "", nil, nil, err
+		return err
 	}
 	// Remove the csv file after creating the zip file
 	err = os.Remove(csvFilePath)
 	if err != nil {
-		return "", nil, nil, err
+		return err
+	}
+	return nil
+}
+
+func (b *BingAdsBulkUploader) CreateZipFile(filePath, audienceId string) ([3]string, [3][]int64, [3][]int64, error) {
+	failedJobIds := [3][]int64{}
+	successJobIds := [3][]int64{}
+
+	textFile, err := os.Open(filePath)
+	if err != nil {
+		return [3]string{"", "", ""}, successJobIds, failedJobIds, err
 	}
+	defer textFile.Close()
 
-	return zipFilePath, successJobIds, failedJobIds, nil
+	if err != nil {
+		return [3]string{"", "", ""}, successJobIds, failedJobIds, err
+	}
+	var csvWriter [3]*csv.Writer
+	var csvFilePaths [3]string
+	var zipFilePaths [3]string
+	var eventCount [3]int
+	csvFilePaths[0], csvWriter[0], zipFilePaths[0] = csvZipFileCreator(audienceId, "Add")
```

Comment:
it is not necessary that always we have all of them everytime

Reply:
@ItsSudip

Reply:
I am creating a placeholder for them only. I already removed the file whenever not required.